packages/medusa/src/types/common.ts

```diff
@@ -14,36 +14,88 @@ export interface FindConfig<Entity> {
 
 export type PaginatedResponse = { limit: number; offset: number; count: number }
 
+export class DateComparisonOperator {
+  @IsDate()
+  lt?: Date | string
+
+  @IsDate()
+  gt?: Date | string
+
+  @IsDate()
+  gte?: Date | string
+
+  @IsDate()
+  lte?: Date | string
+}
+
+export class StringComparisonOperator {
+  @IsString()
+  lt?: string
+
+  @IsString()
+  gt?: string
+
+  @IsString()
+  gte?: string
+
+  @IsString()
+  lte?: string
+}
+
+export class NumericalComparisonOperator {
+  @IsNumber()
+  lt?: number
+
+  @IsNumber()
+  gt?: number
+
+  @IsNumber()
+  gte?: number
+
+  @IsNumber()
+  lte?: number
```

Comment:
The signature for each of the properties should be:
```
@IsNumber()
@Type(() => Number)
lt?: number
```
By including `@Type` from the `class-transformer` package we get the type conversion to a number, this is required for the `@IsNumber` validator to pass, but will also convert the result to a Number
`@IsOptional` is required on the decorator level as well, otherwise users of the API are required to include `lt`, `gt`, `gte`, and `lte` even when only making a request for one of them.

The same should be applicable to the `StringComparisonOperator` but without typeconversion
```suggestion
  @IsNumber()
  @IsOptional()
  @Type(() => Number)
  lt?: number

  @IsNumber()
  @IsOptional()
  @Type(() => Number)
  gt?: number

  @IsNumber()
  @IsOptional()
  @Type(() => Number)
  gte?: number

  @IsNumber()
  @IsOptional()
  @Type(() => Number)
  lte?: number
```